site/content/extend/plugins/signing-plugins.md

```diff
@@ -0,0 +1,159 @@
+
+```console
+> gpg --expert --full-gen-key
+
+Please select what kind of key you want:
+   (1) RSA and RSA (default)
+   (2) DSA and Elgamal
+   (3) DSA (sign only)
+   (4) RSA (sign only)
+   (7) DSA (set your own capabilities)
+   (8) RSA (set your own capabilities)
+   (9) ECC and ECC
+  (10) ECC (sign only)
+  (11) ECC (set your own capabilities)
+  (13) Existing key
+
+Your selection? 9
+
+Please select which elliptic curve you want:
+   (1) Curve 25519
+   (3) NIST P-256
+   (4) NIST P-384
+   (5) NIST P-521
+   (6) Brainpool P-256
+   (7) Brainpool P-384
+   (8) Brainpool P-512
+   (9) secp256k1
+
+Your selection? 1
+
+Please specify how long the key should be valid.
+         0 = key does not expire
+      <n>  = key expires in n days
+      <n>w = key expires in n weeks
+      <n>m = key expires in n months
+      <n>y = key expires in n years
+
+Key is valid for? (0) 0
+
+Key expires at ...
+
+Is this correct? (y/N) y
+
+GnuPG needs to construct a user ID to identify your key.
+
+Real name: Mattermost Inc
+
+Email address: info@mattermost.com
+
+Comment:
+You selected this USER-ID:
+    "Mattermost Inc <info@mattermost.com>"
+
+Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
+```
+
+
+### Exporting the Private Key
+Find the ID of your private key first. The ID is a hexadecimal number.
+```console
+> gpg --list-secret-keys
+```
+This is your private key and you should keep it secret. Replace `XXXXXXXX` with your hexadecimal key ID
+```console
+> gpg --export-secret-keys XXXXXXXX > ./my-priv-key.plugin.gpg
+```
+
+
+### Exporting the Public Key
+Find the ID of your public key first. The ID is a hexadecimal number.
+```console
+> gpg --list-keys
+```
+Replace `XXXXXXXX` with your hexadecimal key ID
+```console
+> gpg --export XXXXXXXX > ./my-pub-key.plugin.gpg
+```
+
+
+### Importing the Key
+If you already have a public and private key pair, you can import them to the GPG.
+```console
+> gpg --import ./my-priv-gpg-key.plugin.gpg
+> gpg --import ./my-pub-gpg-key.plugin.gpg
+```
+
+
+## Plugin Signing
+For plugin signing you have to know the hexadecimal ID of the private key. Let's assume you want to sign `com.mattermost.demo-plugin-0.1.0.tar.gz` file. Replace `XXXXXXXX` with your hexadecimal key ID and run:
+```console
+> gpg -u XXXXXXXX --verbose --personal-digest-preferences SHA256 --detach-sign com.mattermost.demo-plugin-0.1.0.tar.gz
+```
+This command will generate `com.mattermost.demo-plugin-0.1.0.tar.gz.sig` which is the signature of your plugin.
+
+## Plugin Verification
+Mattermost server should be able to verify plugins before install. the public key has to be added on the server. For this purpose, start the Mattermost server and run the following command:
+```console
+> mattermost plugin add key my-pub-key.plugin.gpg
```

Comment:
1/5 that we should not encourage people to use `plugin.gpg` as an extensions, as this is an internal convention.

Reply:
True. We are not. We will automatically add the `plugin.gpg` extension if it's missing.